### DISCUSSION TITLE
Cannot run "rugpi-ctrl update" command

### DISCUSSION THREAD

**Discussion by parth-shah-simform** [silitics/rugix#10] [Q&A] [answered]
We want to provide OTA update to the raspberry pi 4 using rugpi. We have followed steps in this link[https://oss.silitics.com/rugpi/docs/getting-started](url) and created a build/customized.img file. 

- Now. we want to update the system with the .img file created above. 

- But When we run` rugpi-ctrl update install <path to the artifact>` command it gives us error saying `bash: rugpi-ctrl: command not found`. 
 
- So what are the steps to install "rugpi-ctrl" and update the system with the created customized.img

**Comment by koehlma** [comment, marked as answer]
Can you clarify where you tried to run `rugpi-ctrl`?

In general, Rugpi Ctrl is installed into the image by the `rugpi-ctrl` recipe, which is enabled by default. To me, it sounds like you are trying to install an update on a system which has not been provisioned with Rugpi in the first place. Such kinds of updates are generally not possible. You can only update a system which has been provisioned with Rugpi. There is no way to install Rugpi post-hoc on an existing system which has not been built with Rugpi.

**Reply by parth-shah-simform** [reply]
Could you recommend documentation on provisioning a device with Rugpi? I have already checked the links below, but I couldn't find a method to perform this on my current device (Raspberry Pi), which needs to be set up over-the-air (OTA).

Already referred links:
- https://oss.silitics.com/rugpi/docs/getting-started
- https://oss.silitics.com/rugpi/docs/guide/over-the-air-updates

**Reply by koehlma** [reply]
You need to have physical access to the device and flash the image on the SD card, e.g., using Raspberry Pi Imager, as you would with any of the official images. You then simply put the SD card into the Raspberry Pi and boot from it. Only after this initial step you can then install a new image over-the-air, e.g., by using [Rugpi Admin](https://oss.silitics.com/rugpi/docs/guide/rugpi-admin).

**Reply by parth-shah-simform** [reply]
Should we use the Raspbian OS image for flashing, or should we use the build/customized.img generated by run-bakery? The flashing of the build/customized.img didn't yield the expected results.

**Reply by koehlma** [reply]
You must flash `build/customized.img` onto the SD card.

What do you mean by “didn't yield the expected results”?

**Reply by parth-shah-simform** [reply]
After initiating the Raspberry Pi with the recently flashed SD Card, it becomes unresponsive and remains stuck on the screen shown in attached page.

![Error Image](https://github.com/silitics/rugpi/assets/122003466/dfaf3a3f-39be-4469-a40e-1d02e2111c46)

## Our objective is as follows:
We aim to deliver Over-The-Air (OTA) updates for the operating system to numerous Raspberry Pi devices situated in remote locations. Can Rugpi facilitate these OTA updates?

**Reply by koehlma** [reply]
Thanks, indeed there was a bug preventing the firmware update to be included.

Please pull the latest Docker image with
```
docker pull ghcr.io/silitics/rugpi-bakery:latest
```
and build the image again – it should now work.

Regarding your objective: Are you asking whether you can update devices which will be built and provisioned with Rugpi or are these devices already deployed in the field? In case the devices are  new and can be provisioned with Rugpi, sure, you can use Rugpi to do the OTA updates – that's its purpose. In case the devices are already deployed in the field, Rugpi will not help you. Unfortunately, I am also not aware of anything that can update the devices in that case.

**Reply by parth-shah-simform** [reply]
We have pulled the latest Docker image with the above command. After that we have created a new customized image with the same steps present in the docs. However, upon flashing it to the SD Card, it becomes unresponsive and remains stuck on the screen, as depicted in the attached image.

![Selected photo](https://github.com/silitics/rugpi/assets/122003466/bf308e0c-3110-4e0f-a113-3b305f5c03d5)

**Reply by koehlma** [reply]
This is how it is supposed to look, what are you expecting? It booted successfully and you can now access it via SSH.

**Reply by parth-shah-simform** [reply]
We were expecting a GUI after boot. Does this OS not have any GUI?

After accessing via SSH it denies id and password as "pi" and "root" respectively (We have configured these when flashing the raspbian os in Raspberry Pi Imager). Is there any default password present?

**Reply by koehlma** [reply]
You are building an image based on Raspberry Pi OS Lite which does not have a GUI. If you want a GUI, then you have to use the full version of Raspberry Pi OS as a basis – although, I never actually tested that.

For security reasons, Rugpi does not come with any users and passwords by default. You can bake your SSH key into the image, this is all described in the [quick start guide in step 2](https://oss.silitics.com/rugpi/docs/getting-started). The idea is that you use recipes to configure the system to your needs. These recipes can also create users and set their passwords, if that is what you want. Applying any settings with Raspberry Pi Imager, for now, is unsupported. We may add support for that at some point, but for now, it is not a priority. Rugpi is not something you can simply flash and then use in the same way as Raspberry Pi OS. It is meant to for building customized variants of Raspberry Pi OS and anything you want to customize has to be done via recipes.

Anyway, thanks for making me aware that the docs are not sufficiently clear in that Raspberry Pi Imager cannot be used to configure the system, I will fix that.
